src/util/report.py

```diff
@@ -291,6 +291,21 @@ def ref(xid):
     except Exception:
         _logger.warning("Cannot announce message", exc_info=True)
 
+    # Chat window with the report will be open post-upgrade for the admin user
+    if version_between("9.0", "saas~18.2") and user.partner_id and recipient:
+        channel_member_model = (
+            "discuss.channel.member"
+            if version_gte("saas~16.3")
+            else "mail.channel.member"
+            if version_gte("16.0")
+            else "mail.channel.partner"
+        )
+        domain = [("partner_id", "=", user.partner_id.id), ("channel_id", "=", recipient.id)]
+        try:
```

Comment:
Alternatively you can add an else to the try/except block above
```
else:
    # Chat window with the report will be open post-upgrade for the admin user
    if version_between("9.0", "saas~18.2") and user.partner_id and recipient:
        ...
```
If the message was post it's really unexpected that we fail here. Moreover if it failed there is no point in making it visible :/

Reply:
@aj-fuentes why does it need to be in the `else` and not in the first `try` block? We still need this line wrapped in a `try/except`:

`registry[channel_member_model].search(domain)[:1].with_context(ctx).fold_state = "open"`

If we put it in the `else`, then I assume the point is to **not** catch any exception when it's executed?

Reply:
When the posting failed this will fail _always_. That's what I meant. I didn't imply you should remove the other other try/except, you can keep it. In other words, why would we generate two errors in the logs if with the first one is already enough? OTOH if you put in the try/except above you may show a misleading error in the logs since maybe the posting succeeded but it was only the "opening" that failed. (that's what I meant that it is unexpected that this fail if the posting succeeded...)

Reply:
Ah, I understand. I was confused by the use of _alternatively_ there :p